e2e/stack_test.go

```diff
@@ -117,6 +122,46 @@ func TestE2E(t *testing.T) {
 	runningTests.Wait()
 }
 
+func checkForRollbacks(t *testing.T, stack *e2e.StackConfig) {
+	// Subscribe to new block header events
+	const subscriber = "rollbackChecker"
+	eventChan, err := stack.L2Client.Subscribe(stack.Ctx, subscriber, bfttypes.QueryForEvent(bfttypes.EventNewBlockHeader).String())
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, stack.L2Client.UnsubscribeAll(stack.Ctx, subscriber))
+	}()
+
+	l1BlockTicker := time.NewTicker(250 * time.Millisecond)
+	defer l1BlockTicker.Stop()
+
+	var lastBlockHeight int64
+
+	for {
+		select {
+		// Check new block header events for rollbacks
+		case event := <-eventChan:
+			eventNewBlockHeader, ok := event.Data.(bfttypes.EventDataNewBlockHeader)
+			require.True(t, ok)
+			currentHeight := eventNewBlockHeader.Header.Height
+
+			// Ensure that the current block height is not less than the last checked height
+			if lastBlockHeight > 0 && currentHeight < lastBlockHeight {
+				require.FailNow(t, "monomer has rolled back")
+			}
+			lastBlockHeight = currentHeight
+
+		// End the test once a sequencing epoch has passed.
+		case <-l1BlockTicker.C:
```

Comment:
Rather than using a ticker, can we check the L1Info on the block to determine the epoch? This also means we won't need to add the `SequencingWindowSize` to the `StackConfig` struct.

Reply:
Are you referring to using the `x/rollup` query endpoint to get the L1 block info or just extracting the L1 info from the L1 attributes tx?

The query endpoint still isn't implemented, that work is tracked by [this issue](https://github.com/polymerdao/monomer/issues/189).

Reply:
I mean checking the first transaction in the block.